dashboard/app/models/course.rb

```diff
@@ -18,6 +18,15 @@ class Course < ApplicationRecord
   has_one :plc_course, class_name: 'Plc::Course'
   has_many :course_scripts, -> {order('position ASC')}
 
+  validates :name,
+    presence: true,
+    uniqueness: {case_sensitive: false},
+    format: {
+      unless: :plc_course,
+      with: /\A[a-z0-9\-]+\z/,
+      message: 'can only contain lowercase letters, numbers and dashes'
+    }
```

Comment:
Using a concern to do this may be easier to do than I thought. I will explore doing this in a follow on PR